### PR TITLE
Allow using integer vectors for iteration and make range() use them

### LINKED ISSUE #34121: range(x,y) malfunctions for values greater than 0x1000000
**Godot version:**
Godot_v3.1.2-stable_win64

**Issue description:**
Using range(x,y) with values greater than 0x1000000 (16777216), where x and/or y are odd, results in an incorrect array with one or two extra values (as if x were x-1 and/or y were y+1).

Here is a code snippet to show this:
```
   for i in range(16777213, 16777215):
      print(i)
   print("====")
   for i in range(16777217, 16777219):
      print(i)
   print("====")
```

Its output is:
```
16777213
16777214
====
16777216
16777217
16777218
16777219
====
```

The former is correct, the latter is wrong.

P.S. The bug is slightly more complex than I had described. Not only may it result in one or two extra values; it may also result in one or two missing values. For example, range(16777219, 16777221) returns an empty array.

P.P.S. This bug also affects the range(x,y,step) flavour.

### PR DESCRIPTION
**Pull request by aaronfranke**
This fixes #34121. Since it causes failures with large values, and because floats aren't ideal for integer iteration anyway, we don't want to optimize range's parameters into Vector2/Vector3. Instead, let's use Vector2i/Vector3i which have a larger range (and might perform better).

Using Vector2i/Vector3i still has a limit at 2^31, which is lower than what normal GDScript ints allow (2^63), but I tried removing the optimization entirely and it still fails at 2^31 anyway. If it's ever desired to go above 2^31, that can wait for later, and I don't think I'll be the one doing that.

Using Vector2i/Vector3i as iterators also means that we can directly use them instead of the word "range", which may be useful if you want to store a range in a variable.

Here's some test code:

```gdscript
	for i in Vector2(16777213, 16777221):
		print(i) # Fails at the last value because 16777221 becomes 16777220.
		# At least, this will fail as long as real_t is float, but it would work when real_t is double.
	print("====")
	for i in Vector2i(16777213, 16777221):
		print(i) # Works.
	print("====")
	for i in range(16777213, 16777221):
		print(i) # Works.
	print("====")
```

I also changed a function that returns a `bool` value so that its return type is also `bool`, used `int64_t` in a few places where the end result becomes `int64_t` anyway, and I did a little formatting in the beginning of `variant_op.cpp` because it wasn't aligned before and it was ugly.

### CODE REVIEW

**Comment by bojidar-bg** [comment]
What about changing Vector2/3 iteration to use float start/end/increment, instead of casting them to int (thus reverting #8783)?
That would allow for things like `for angle in Vector3(0.0, 2 * PI, PI / 10)`

**Comment by Calinou** [comment]
> Using Vector2i/Vector3i as iterators also means that we can directly use them instead of the word "range", which may be useful if you want to store a range in a variable.

IIRC, the result of `range()` can be stored into a variable, as it's just an array. That said, it will be slow when creating large arrays (even if you don't store its result into a variable).

**Comment by Calinou** [comment]
Does using `Vector2i()` instead of `range()` avoid allocating a GDScript array? If so, this can perform much better when iterating over a large range. We should make sure to document this :slightly_smiling_face:

**Comment by bojidar-bg** [comment]
@Calinou `for i in range(x,y)` has been converted to `for i in Vector2(x, y)` ever since bfef8de1bc4f7a7b9617a7b181881129033a0b0e

**Comment by aaronfranke** [comment]
@bojidar-bg That's a great idea! I just pushed a 2nd commit. Example code:

```gdscript
	for angle in Vector3(0.0, 2 * PI, PI / 10):
		print(angle)
```

Output:

```
0
0.314159
0.628319
0.942478
1.256637
1.570796
1.884956
2.199115
2.513274
2.827433
3.141593
3.455752
3.769911
4.084071
4.39823
4.712389
5.026548
5.340708
5.654867
5.96902
```

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -8774,7 +8774,7 @@ int GDScriptParser::get_completion_argument_index() {
 	return completion_argument;
 }
 
-int GDScriptParser::get_completion_identifier_is_function() {
+bool GDScriptParser::get_completion_identifier_is_function() {
```
Comment (vnen) [inline modules/gdscript/gdscript_parser.cpp:8777]: Why is this change here? I get that `bool` makes more sense, but it's not related to the other changes.

Comment (aaronfranke) [reply modules/gdscript/gdscript_parser.cpp:8777]: Because it's a minor fix that I noticed while making this PR and it's easier than making a separate PR just for this one change, but I could do that if you want.

Comment (vnen) [reply modules/gdscript/gdscript_parser.cpp:8777]: At least do it in a different commit, so the blame is more precise.

Comment (aaronfranke) [reply modules/gdscript/gdscript_parser.cpp:8787]: Done

**Review by vnen — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
